### PR TITLE
Add support for __imatmul__ wiring

### PR DESCRIPTION
**Pull request by cdonovick**

### CODE REVIEW

**Comment by cdonovick** [comment]
implements #495

File: magma/t.py [hunk 1]
```diff
@@ -73,6 +73,7 @@ def __le__(self, other):
         else:
             raise TypeError(f"Cannot use <= to assign to output: {self.debug_name} (trying to assign {other.debug_name})")
```
Comment (leonardt) [inline magma/t.py]: Should we update this error message? (And maybe add a deprecation warning? Or maybe wait to do this)

Comment (cdonovick) [reply magma/t.py]: Heh was going to add another commit to address the error message but then decided I didn't actually want to write new code.   I will fix.

Deprecation warning is up to you.

Comment (leonardt) [reply magma/t.py]: I'll add it as part of the new release after hwtypes2 is merged

**Pull request merged**